com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -129,6 +133,15 @@ void UpgradeObjectReferences(GameObject[] rootObjects)
                     }
                     return oldValue;
                 });
+                
+                // manager cameras need to be recorded to keep references after saving scenes if they are part of a prefab
+                if (PrefabUtility.IsPartOfAnyPrefab(go))
```

Comment:
I moved the code to RecursiveUpdateBehaviourReferences, because the lambda does not modify the value, it just gets the new value. The value is set based on the result of ScanFields in RecursiveUpdateBehaviourReferences.